tests/mocha/json_test.js

```diff
@@ -0,0 +1,343 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+suite('JSON', function() {
+  setup(function() {
+    this.workspace_ = new Blockly.Workspace();
+    this.blocks_ = [];
+    this.blockTypes_ = [];
+    this.messages_ = [];
+  });
+
+  teardown(function() {
+    for (var i = 0; i < this.blocks_.length; i++) {
+      var block = this.blocks_[i];
+      block.dispose();
+    }
+    for (var i = 0, blockType; (blockType = this.blockTypes_[i]); i++) {
+      delete Blockly.Blocks[blockType];
+    }
+    for (var i = 0, message; (message = this.messages_[i]); i++) {
+      delete Blockly.Msg[message];
+    }
+    this.workspace_.dispose();
```

Comment:
Not really for this PR, but might be good to see if this is needed in other tests. If so, would be good to pull out into a function for use throughout all tests.

Reply:
I'm working out what exactly this should look like but I'm fairly certain I'm going to add this functionality as a helper function that all the tests can call.

Reply:
For future PR.